ui/builder.js

```diff
@@ -0,0 +1,188 @@
+const fs = require('fs-extra')
+const os = require('os')
+const { start } = require('live-server')
+const { createProxyMiddleware } = require('http-proxy-middleware')
+const { watch } = require('chokidar')
+const { build } = require('esbuild')
+// TODO: submit PR for upstream
+fs.copyFileSync(
+  './postcss2-index.js',
+  './node_modules/esbuild-plugin-postcss2/dist/index.js'
+)
+const postCssPlugin = require('esbuild-plugin-postcss2')
+const { yamlPlugin } = require('esbuild-plugin-yaml')
+const svgrPlugin = require('esbuild-plugin-svgr')
+const logTime = require('./esbuild/plugins/logtime')
+
+const isDev = process.env.NODE_ENV !== 'production'
+
+// handle .env
+if (isDev) {
+  fs.copyFileSync('./.env.development', './.env')
+} else {
+  fs.copyFileSync('./.env.production', './.env')
+}
+// load .env file
+require('dotenv').config()
+
+const dashboardApiPrefix =
+  process.env.REACT_APP_DASHBOARD_API_URL || 'http://127.0.0.1:12333'
+const devServerPort = process.env.PORT
+const devServerParams = {
+  port: devServerPort,
+  root: 'build',
```

Comment:
Maybe we should define a global var `const outputDir = 'build'` instead of hard-code.